src/fswAlgorithms/opticalNavigation/cobConverter/cobConverter.cpp

```diff
@@ -173,6 +178,33 @@ void CobConverter::UpdateState(uint64_t CurrentSimNanos)
         Eigen::Matrix3d covar_N = dcm_BN.transpose() * covar_B * dcm_BN;
         Eigen::Matrix3d covar_C = dcm_CB.transpose() * covar_B * dcm_CB;
 
+        bool goodOutlierCheck = true;
+        if (this->performOutlierDetection){
+            FilterMsgPayload filterMsgBuffer = this->opnavFilterInMsg();
+
+            int numberOfStates = filterMsgBuffer.numberOfStates;
+            Eigen::VectorXd filterState = cArray2EigenMatrixXd(filterMsgBuffer.state, numberOfStates, 1);
+            Eigen::Vector3d rNav_BN_N = filterState.segment(0, 3);
+            Eigen::Vector3d rhatNav_N = rNav_BN_N.normalized();
+            Eigen::MatrixXd filterCovariance = cArray2EigenMatrixXd(filterMsgBuffer.covar,
+                                                                    numberOfStates,
+                                                                    numberOfStates);
+            Eigen::Matrix3d covarNav_N = filterCovariance.block(0, 0, 3, 3) / pow(rNav_BN_N.norm(), 2);
+
+            Eigen::Matrix3d covarImage = computeTotalCobCovariance(covarNav_N,
+                                                                   this->covarAtt_BN_B,
+                                                                   covarCob_C,
+                                                                   dcm_NC.transpose(),
+                                                                   dcm_CB,
+                                                                   cameraCalibrationMatrix);
+            goodOutlierCheck = cobOutlierDetection(rhatCOB_C,
+                                                   rhatNav_N,
+                                                   covarImage,
+                                                   dcm_NC.transpose(),
+                                                   cameraCalibrationMatrix,
+                                                   this->numStandardDeviations);
```

Comment:
I'm debating if we can not compute the covarImage if the setStandardDeviation is true. Just to save some unused computations

Reply:
Definitely worth a refactor, considering `this->numStandardDeviations` is being passed into a member function.

Reply:
Added an if statement such that the covariance is only computed if `specifiedStandardDeviation` is False.

Reply:
Also removed `numStandardDeviations ` from the input parameters.